src/qt/bitcoin.cpp

```diff
@@ -388,14 +388,16 @@ void BitcoinApplication::requestShutdown()
     pollShutdownTimer->stop();
 
 #ifdef ENABLE_WALLET
-    // Delete wallet controller here manually, instead of relying on Qt object
+    // Delete splash screen and wallet controller here manually, instead of relying on Qt object
     // tracking (https://doc.qt.io/qt-5/objecttrees.html). This makes sure
     // walletmodel m_handle_* notification handlers are deleted before wallets
     // are unloaded, which can simplify wallet implementations. It also avoids
     // these notifications having to be handled while GUI objects are being
     // destroyed, making GUI code less fragile as well.
     delete m_wallet_controller;
     m_wallet_controller = nullptr;
+    delete m_splash;
+    m_splash = nullptr;
```

Comment:
> the object will not be deleted if a shutdown is requested during load.

It will, via Qt a parent-child relation.

Early widget deletion is crucial when wallet code is active only.